Development_v2

ciaranparker opened the pull request:
An Object Oriented C# game where the player enters a dungeon and battles an enemy. This version includes improvements which have been implemented after taking into account previous code review. Error checking has been significantly improved, any invalid user input are recognised as exceptions and handled resulting in an appropriate output, rather than crashing the game or allowing the user to continue. I have also included break statements and made certain booleans false after certain code. This prevents the code from endlessly looping. Now when a player or enemy dies the game can be closed and exited by the user.

CAusten1337 commented on Game.cs:
```diff
@@ -1,25 +1,219 @@
+
+                    if (playerUsername.Length == 0)
+                    {
+                        throw new NullReferenceException("Username cannot be empty");
```
Probs should have used "ArgumentNullException" instead of the NullReferenceException. This is just because the exception relates to the specific argument rather than like an object that doesn't exist

CAusten1337 commented on Game.cs:
```diff
@@ -1,25 +1,219 @@
-            bool playing = false;
+            Console.WriteLine("Entering Dungeon...");
+            Console.WriteLine(currentRoom.GetDescription());
+            Console.WriteLine("You can enter 'M' to view the description at any time");
```
Could always make this one line but would look horrible and messy

CAusten1337 commented on Game.cs:
```diff
@@ -1,25 +1,219 @@
+                            lootingRoom = false;
+                        }
+
+                        else if (userMoveInput == "M")
```
The player has no idea this option exists

CAusten1337 commented on Player.cs:
```diff
@@ -1,25 +1,47 @@
+            Console.WriteLine($"Health: {Health}");
+        }
+
+        
```
Why is there so much empty space here?

CAusten1337 commented on Program.cs:
```diff
@@ -10,11 +10,24 @@ internal class Program
+                Console.WriteLine("Waiting for your Implementation");
+                Console.WriteLine("Press any key to exit...");
+                Console.ReadKey();
+            }
```
love the use of the "try" to allow for this to work so well

CAusten1337 reviewed (commented):
Looks lovely, reads well.
A true pleasure of code to enjoy.

JamesBowman-06 commented:
There are quite a few empty lines, also you should reduce the length of the lines making it easier to read. You could also make lines display one by one rather than all at once, making it easier to read. However the code meets the criteria which have been outlined.

ciaranparker commented:
I have deleted any unnecessary comments or empty lines to make the code less overwhelming and therefore making it easier to read. I have reduced the length of some of the longer lines splitting some of the code into 2 lines so that users do not have to have to read one whole line because this could be overwhelming. While I could not make the code display in the terminal 1 by one I split some of the lines into 2 or 3 lines so that it is not as overwhelming to read. I also changed the items aspect, putting items in the room class so that they can be defined and access in the game class. This will ensure that different rooms can have items and also makes it easier to reuse these items throughout the code. I have also added logical or in if statements so that the code can still run if the user puts a lowercase character instead of the uppercase, to avoid frustration for the user.

I also changed the exception on line 53 from NullReferenceException to ArgumentNullException as advised due to it referring to a specific argument.
